modules/transactions/src/main/java/org/apache/ignite/internal/tx/InternalTransaction.java

```diff
@@ -143,4 +144,17 @@ IgniteBiTuple<ClusterNode, Long> enlist(
      * @return The future.
      */
     CompletableFuture<Void> kill();
+
+    /**
+     * Rolls back the transaction if the timeout is exceeded.
+     *
+     * @return The future.
+     */
+    default CompletableFuture<Void> rollbackTimeoutExceededAsync() {
```

Comment:
Please explain why do we need this method in addition to common rollbackAsync().

Reply:
Ive removed the default implementation. I can't see another way to provide an information that the transactions was rolled back because of timeout exceeded.

Reply:
@sanpwc could you please take a look?

Reply:
What will happen if this method is called on already finished transaction?

Reply:
Added a javadoc here: 

```
A rollback of a completed or ending transaction has no effect and always succeeds when the transaction is completed.